dds_web/web/user.py

```diff
@@ -448,15 +448,52 @@ def password_reset_completed():
         return flask.redirect(flask.url_for("auth_blueprint.index"))
 
     units_to_contact = {}
+    unit_admins_to_contact = {}
+
     if user.role != "Super Admin":
         for project in user.projects:
-            if project.responsible_unit.external_display_name not in units_to_contact:
-                units_to_contact[
-                    project.responsible_unit.external_display_name
-                ] = project.responsible_unit.contact_email
-        return flask.render_template(
-            "user/password_reset_completed.html", units_to_contact=units_to_contact
-        )
+            if user.role == "Unit Admin":
+                users = (
+                    db.session.query(models.User)
+                    .join(models.UnitUser)
+                    .join(models.Email)
+                    .with_entities(
+                        models.User.username,
+                        models.User.name,
+                        models.UnitUser.unit_id,
+                        models.UnitUser.is_admin,
+                        models.Email.email,
+                    )
+                    .filter(models.UnitUser.unit_id == user.unit_id)
+                    .filter(models.UnitUser.is_admin == True)
+                    .filter(models.User.username != user.username)
+                    .all()
+                )
+                email = (
+                    db.session.query(models.Email)
+                    .with_entities(models.Email.email)
+                    .filter(models.Email.user_id == user.username)
+                    .first()
+                )
+
+                unit_admins_to_contact = users
+            else:
+                if project.responsible_unit.external_display_name not in units_to_contact:
+                    units_to_contact[
+                        project.responsible_unit.external_display_name
+                    ] = project.responsible_unit.contact_email
+
+        if len(units_to_contact) > 0:
+            return flask.render_template(
+                "user/password_reset_completed.html", units_to_contact=units_to_contact
+            )
+
+        if len(unit_admins_to_contact) > 0:
```

Comment:
You should be able to remove this, and instead pass in `units_to_contact` on line 498, and then make sure that the template checks `{% if units_to_contact %}` before printing anything